ts/src/exkoin.ts

```diff
@@ -0,0 +1,1732 @@
+     * @param {string} [code] unified currency code
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @returns {object} a [ledger entry structure]{@link https://docs.ccxt.com/#/?id=ledger}
+     */
+    async fetchLedgerEntry (id: string, code: Str = undefined, params = {}): Promise<LedgerEntry> {
+        await this.loadMarkets ();
+        const request: Dict = {
+            'id': id,
+        };
+        const response = await this.privateGetPrivateLedger (this.extend (request, params));
+        //
+        // {
+        //     "id": "ledger123",
+        //     "type": "trade",
+        //     "in": true,
+        //     "amount": 100.5,
+        //     "currency": "USDT",
+        //     "created_at": 1729130040,
+        //     "after": 1000.5,
+        //     "ref": "order123"
+        // }
+        //
+        return this._parseLedgerEntry (response);
+    }
+
+    _parseLedgerEntry (item: Dict): LedgerEntry {
+        //
+        // {
+        //     "id": "ledger123",
+        //     "type": "trade",
+        //     "in": true,
+        //     "amount": 100.5,
+        //     "currency": "USDT",
+        //     "created_at": 1729130040,
+        //     "after": 1000.5, // or undefined if originating from time sensitive operation (order-fill, ...)
+        //     "ref": "order123"
+        // }
+        //
+        const timestamp = this._exkoin_safeTimestamp (item, 'created_at');
+        const currencyId = this.safeString (item, 'currency');
+        const currency = this.safeCurrency (currencyId);
+        const amount = this.safeNumber (item, 'amount');
+        const isIn = this.safeBool (item, 'in');
+        const direction = isIn ? 'in' : 'out';
+        const type = this.safeString (item, 'type');
+        const after = this.safeNumber (item, 'after');
+        let before: Num = undefined;
+        if (after !== undefined) {
+            if (direction === 'in') {
+                before = after - amount;
+            } else {
+                before = after + amount;
+            }
+        }
+        const referenceId = this.safeString (item, 'ref');
+        return this.safeLedgerEntry ({
+            'id': this.safeString (item, 'id'),
+            'direction': direction,
+            'account': undefined,
+            'referenceId': referenceId,
+            'referenceAccount': undefined,
+            'type': type,
+            'currency': currency['code'],
+            'amount': amount,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'before': before,
+            'after': after,
+            'status': 'ok',
+            'fee': undefined,
+            'info': item,
+        }, currency);
+    }
+
+    _exkoin_safeTimestamp (item: Dict, key: IndexType): Int {
+        return this.safeInteger (item, key);
+    }
+
+    _parseLedger (data, since: Int = undefined, limit: Int = undefined, params = {}): LedgerEntry[] {
```

Comment:
parserMethods should follow the standard naming, in this case `parseLedgerEntry`  and use the base `parseLedger`